src/components/Header/Header.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { memo } from 'react';
+import cn from 'classnames';
+
+interface HeaderProps {
+  title: string;
+  isLoading: boolean;
+  hasSomeActiveTodos : boolean;
+  onToggleAllTodos: () => void
+  onAddTodo: (title: string) => void;
+  onChangeTitle: (title: string) => void;
+  onChangeNotification: (errorMessage: string) => void;
+}
+
+export const Header: React.FC<HeaderProps> = memo(({
+  title,
+  isLoading,
+  hasSomeActiveTodos,
+  onToggleAllTodos,
+  onAddTodo,
+  onChangeTitle,
+  onChangeNotification,
+}) => {
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onChangeTitle(event.target.value);
+  };
+
+  const handleToggleTodos = () => {
+    onToggleAllTodos();
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      onChangeNotification('Title can\'t be empty');
+
+      return;
+    }
+
+    onAddTodo(title);
+  };
+
+  return (
+    <header className="todoapp__header">
+      {true && (
+        // eslint-disable-next-line jsx-a11y/control-has-associated-label
```

Comment:
The `eslint-disable-next-line jsx-a11y/control-has-associated-label` is used to bypass accessibility requirements for controls. Consider providing an accessible label for the button.